### DISCUSSION TITLE
Q: What params are used to configure python?

### DISCUSSION THREAD

**Discussion by apivovarov** [jdx/mise#2107] [Q&A]
Where can I found what Python configure params are used when I run command `rtx use -g python@3.9`?

I guess it should be smth like `./configure --enable-optimizations --with-lto etc etc`.

[pyenv/plugins/python-build](https://github.com/pyenv/pyenv/tree/master/plugins/python-build) expects it as `PYTHON_CONFIGURE_OPTS`

**Comment by jdx** [comment]
It just calls python-build

**Comment by apivovarov** [comment]
What should I do to build python with optimizations using rtx? 
Should I do the following?
1. `export PYTHON_CONFIGURE_OPTS="--enable-optimizations --with-lto"`
2. `rtx uninstall python@3.9`
3. `rtx use -g python@3.9`

**Comment by jdx** [comment]
yeah but `PYTHON_CONFIGURE_OPTS="--enable-optimizations --with-lto" rtx i python@3.9 -f` will also do it

**Comment by JamBalaya56562** [comment]
The mechanism in the answer above is still right, but **the default changed and the recipe no longer works as written.**

mise installs **precompiled** python by default now, and `PYTHON_CONFIGURE_OPTS` is a python-build variable — if python-build never runs, exporting it does nothing at all, silently. Measured on **v2026.8.3**:

```console
$ mise install python@3.13.5
mise python@3.13.5   [1/3] download cpython-3.13.5+20250723-x86_64-unknown-linux-gnu-install_only_stripped.tar.gz
mise python@3.13.5   [1/3] verify GitHub artifact attestations
mise python@3.13.5   [1/3] ✓ GitHub artifact attestations verified
mise python@3.13.5   [1/3] extract cpython-3.13.5+20250723-x86_64-unknown-linux-gnu-install_only_stripped.tar.gz
```

No `./configure` anywhere.

`python.compile` is what decides, and it is tri-state:

- `true` — always compile with python-build
- `false` — always download a precompiled binary
- **unset (the default)** — precompiled if one exists for your platform, compile otherwise

So on a platform that has precompiled builds — which is most of them — you need the setting first:

```toml
[settings]
python.compile = true
```

With that in place the original answer holds. Verified by passing a deliberately invalid flag, so the failure proves the variable reached `./configure` rather than being ignored:

```console
$ PYTHON_CONFIGURE_OPTS="--bogus-flag-xyz" mise install -f python@3.13.5
-> https://www.python.org/ftp/python/3.13.5/Python-3.13.5.tar.xz
Installing Python-3.13.5...
BUILD FAILED (Ubuntu 24.04 using python-build 2.8.3-6-g17be3086)
...
configure: error: unrecognized option: `--bogus-flag-xyz'
```

Swap in the real options and it builds with them. `MISE_PYTHON_COMPILE=1` works for a one-off instead of the setting, and `-f` still means "force reinstall even if already installed".

One related knob if you are after performance rather than specific flags: `python.precompiled_arch` chooses which precompiled build to fetch (defaults to `x86_64_v3` / `aarch64`), so an older CPU may want `x86_64` for compatibility — a way to influence the binary without compiling at all.
